aptos-move/framework/src/natives/aggregator_natives/helpers_v2.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_aggregator::aggregator_extension::AggregatorID;
+use move_binary_format::errors::PartialVMResult;
+use move_vm_types::values::{Reference, StructRef, Value};
+
+/// Indices of `value` and `limit` fields in the `Aggregator` Move
+/// struct.
+const VALUE_FIELD_INDEX: usize = 0;
+const LIMIT_FIELD_INDEX: usize = 1;
+
+/// Given a reference to `Aggregator` Move struct returns a field value at `index`.
+pub(crate) fn get_aggregator_field(aggregator: &StructRef, index: usize) -> PartialVMResult<Value> {
+    let field_ref = aggregator.borrow_field(index)?.value_as::<Reference>()?;
+    field_ref.read_ref()
+}
+
+/// Returns ID and a limit of aggrgegator based on a reference to `Aggregator` Move struct.
+pub(crate) fn aggregator_info_u128(
+    aggregator: &StructRef,
+) -> PartialVMResult<(AggregatorID, u128)> {
+    let (value, limit) = get_aggregator_fields_u128(aggregator)?;
+    assert!(
```

Comment:
Assertion seems ok, but given it is used by natives I would rather have an `Err` here. @gelash what do you think?

Reply:
I added a new error code 0x2_0008 and returning this error code for this.